### PR TITLE
[RLlib] Entropy coeff schedule bug fix and git bisect script.

### LINKED ISSUE #15936: [RLlib] IMPALA (any tf algo using entropy coeff) slowdown observed
Since ray 1.2, IMPALA (and any other tf algorithm that uses `entropy_coeff`) has slowed down due to a bug.
- The bug is caused by a tf-(static graph)-op being added to the graph each time we call `on_global_var_update` (defined inside the `EntropyCoeffSchedule` (tf) mixin class).

To reproduce:

`rllib train -f rllib/tuned_examples/impala/pong-impala-fast.yaml`

Runs much slower than in ray<=1.1.

### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by sven1977**
[RLlib] Entropy coeff schedule bug fix and git bisect script.
- Also moves the existing git bisect script into the `tests` folder and enhances it so it handles stopping, uninstalling, re-installing (the current bisect'd commit), and re-starting ray all before the actual test is run. The test can now also cover whether an algorithm runs fast enough (ignoring the reward).

See: https://github.com/ray-project/ray/issues/15936

Issue #15936 
## Why are these changes needed?

Closes #15936
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/policy/tf_policy.py [hunk 1]
```diff
@@ -983,35 +983,47 @@ class EntropyCoeffSchedule:
 
     @DeveloperAPI
     def __init__(self, entropy_coeff, entropy_coeff_schedule):
-        self.entropy_coeff = get_variable(
-            entropy_coeff,
-            framework="tf",
-            tf_name="entropy_coeff",
-            trainable=False)
-
+        self._entropy_coeff_schedule = None
```
Comment (richardliaw) [inline rllib/policy/tf_policy.py:986]: Should you add a test for this?

Comment (sven1977) [reply rllib/policy/tf_policy.py:986]: You mean for the mixin class? Also for `LearningRateSchedule`? Sure.
We do have extensive tests for the different `Schedule` classes.

Comment (sven1977) [reply rllib/policy/tf_policy.py:987]: done

Comment (sven1977) [reply rllib/policy/tf_policy.py:987]: Added extensive schedule testing (within a Policy) to test_ppo.py. These will test tf/tf2/torch policies for their handling of different lr- and entropy-coeff schedules.

**Review by richardliaw — APPROVED** [review]
